### PR TITLE
Bug fixes and improvements for better adapter support

### PR DESCRIPTION
**Pull request by arobson**
* Support metric name differentiation when using the instrument call
 * Support conversion between byte units
 * Add name to metadata for all system utilization metrics
 * Bug fix - missing require for when in index

### CODE REVIEW

File: spec/behavior/converter.spec.js [hunk 1]
```diff
@@ -3,60 +3,107 @@ var convert = require( '../../convert' );
...
+
+		it( 'should conver terabytes to gigabytes', function() {
+			convert( 3, 'tb', 'gb' ).should.equal( 3072 );
+		} );
+
+		it( 'should conver terabytes to megabytes', function() {
+			convert( 4, 'tb', 'mb' ).should.equal( 4194304 );
+		} );
+
+		it( 'should conver terabytes to kilobytes', function() {
```
Comment (brianedgerton) [inline spec/behavior/converter.spec.js]: Minor misspelling in this test case and the two above it.

**Comment by brianedgerton** [comment]
@arobson Tests pass. Do you want to fix the misspellings or have me merge as is?

**Comment by arobson** [comment]
> OH LOOK AT ME, I'M @brianedgerton AND I CAN SPELL GOODER THAN ALEX CAN BECAUSE ALEX IS A MORON!
-- @brianedgerton on all my PR reviews because I'm a moron.

**Comment by brianedgerton** [comment]
I haved ranned them tests and all be good. :100:

**Pull request merged**